packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -710,24 +856,54 @@ const runGenerator = (context, {outputDir, templateVersion, verbose}) => {
         const assetsDir = p.join(ASSETS_TEMPLATES_DIR, id)
         if (sh.test('-e', assetsDir)) {
             getFiles(assetsDir)
-                .map((file) => file.replace(assetsDir, ''))
-                .forEach((relFilePath) =>
+                .map((file) => {
+                    const relFilePath = file.replace(assetsDir, '')
+                    return relFilePath
+                })
+                .forEach((relFilePath) => {
                     processTemplate(relFilePath, assetsDir, outputDir, context)
-                )
+                })
         }
 
-        // Update the generated projects version. NOTE: For bootstrapped projects this
+        // Process selected Application Extensions
+        processAppExtensions(selectedAppExtensions, extractAppExtensions, appExtensionsDir)
+
+        // Update the generated project's package.json. NOTE: For bootstrapped projects this
         // can be done in the template building. But since we have two types of project builds,
         // (bootstrap/bundle) we'll do it here where it works in both scenarios.
         const pkgJsonPath = p.resolve(outputDir, 'package.json')
         const pkgJSON = readJson(pkgJsonPath)
+
+        // Add selected Application Extensions to devDependencies and mobify object
+        const appExtensionDeps = selectedAppExtensions.reduce((acc, appExtensionName) => {
+            // Find the corresponding Application Extension details
+            const appExtension = context.appExtensions.find((ext) => ext.value === appExtensionName)
+            const version = appExtension ? appExtension.version : '1.0.0-dev'
+
+            acc[`@salesforce/${appExtensionName}`] = extractAppExtensions
+                ? `file:./app/application-extensions/${appExtensionName}`
+                : version
+            return acc
+        }, {})
+
         const finalPkgData = merge(pkgJSON, {
             name: slugifyName(context.answers.project.name || context.preset.id),
-            version: GENERATED_PROJECT_VERSION
+            version: GENERATED_PROJECT_VERSION,
+            devDependencies: appExtensionDeps,
+            mobify: {
+                app: {
+                    extensions: selectedAppExtensions.map((appExtensionName) => [
+                        `@salesforce/${appExtensionName}`,
```

Comment:
This configuration object is obviously only going to work for this specific extension. We probably want to make this configurable, but we can probably leave it until another PR. so an action item for this comment would be to create a new ticket and that new ticket is going to be colocating questions with extension projects so that we can ask them get answers and add them to the configuration object in the tuple for now, you probably just want to have the name of the extension and not the extended, syntax that allows you to provide configuration